lib/idempotency.rb

```diff
@@ -21,4 +22,66 @@ module Idempotency
   def self.cache
     @cache ||= Cache.new(config:)
   end
+
+  def self.use_cache(request, request_identifiers, lock_duration: nil) # rubocop:disable Metrics/AbcSize
```

Comment:
Could we use something like this:

```ruby
def self.use_cache(request, request_identifiers, lock_duration: nil)
  new.use_cache(request, request_identifiers, lock_duration: nil)
end
```

This will allow us to wrap all other methods in private scope, and make test easier by mocking `Idempotency.cache` and `Idempotency.config` using dependencies injection.